pkgs/build-support/fetchzip/default.nix

```diff
@@ -48,8 +48,15 @@
     + extraPostFetch
     # Remove write permissions for files unpacked with write bits set
     # Fixes https://github.com/NixOS/nixpkgs/issues/38649
+    #
+    # However, we should (for the moment) retain write permission on the directory
+    # itself, to avoid tickling https://github.com/NixOS/nix/issues/4295 in
+    # single-user Nix installations. This is because in sandbox mode we'll try to
+    # move the path, and if we don't have write permissions on the directory,
+    # then we can't update the ".." entry.
```

Comment:
Not sure what you mean by ".." directory.
I think the comment is good but should it be spelled out more clearly?

Reply:
Suggestions welcome here! I mean the ".." entry in the directory (i.e. the one that points at the parent directory in standard UNIX parlance)

Reply:
Hmm it must work special in a sandbox?

```bash
❯ mkdir example

❯ cd example

❯ ls -la
total 6420
drwxr-xr-x   2 fmzakari primarygroup    4096 Dec  3 20:02 .
drwxrwxrwt 569 root     root         6565888 Dec  3 20:02 ..

❯ cd ..

❯ chmod -R a-w example

❯ cd ..

❯ /bin/ls -la
total 6420
dr-xr-xr-x   2 fmzakari primarygroup    4096 Dec  3 20:02 .
drwxrwxrwt 569 root     root         6565888 Dec  3 20:03 ..

❯ cd ..

❯ /bin/ls -la  | grep example
dr-xr-xr-x   2 fmzakari   primarygroup    4096 Dec  3 20:02 example

❯ mv example example2

❯ /bin/ls -la  | grep example2
dr-xr-xr-x   2 fmzakari   primarygroup    4096 Dec  3 20:02 example2
```

I am not sure how to reproduce.
FWIW though this fixed my failed derivation before.

Reply:
Here's a reproducer:

```
mkdir -p /tmp/demonstrate/this/thing
chmod a-w /tmp/demonstrate/this/thing

# This will fail
mv /tmp/demonstrate/this/thing /tmp/demonstrate/thing

chmod u+w /tmp/demonstrate/this/thing

# This will succeed
mv /tmp/demonstrate/this/thing /tmp/demonstrate/thing
```

Note that you must move "thing" out of the directory it's in - otherwise the parent directory is still the same, so you don't need to update "..".